lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -880,7 +898,54 @@ def test_dataset_sort(self, server_config, query_as, sort, results):
         for i, d in enumerate(all):
             odd = i & 1
             Metadata.setvalue(d, "global.test.sequence", i)
+            Metadata.setvalue(d, "global.test.mcguffin", 100 - i)
             Metadata.setvalue(d, "user.test.odd", odd, user=test)
         query = {"sort": sort, "metadata": ["dataset.uploaded"]}
         result = query_as(query, "test", HTTPStatus.OK)
         self.compare_results(result.json, results, query, server_config)
+
+    @pytest.mark.parametrize(
+        "sort,message",
+        [
+            (
+                # Specify a sort by a Dataset table column that doesn't exist
+                "dataset.noname",
+                "Metadata key 'dataset.noname' is not supported",
+            ),
+            (
+                # Specify a sort using an undefined order keyword
+                "dataset.name:backwards",
+                "The sort order in 'dataset.name:backwards' must be 'asc' or 'desc'",
+            ),
+            (
+                # Specify a sort using bad sort order syntax
+                "dataset.name:desc:",
+                "The sort order in 'dataset.name:desc:' must be 'asc' or 'desc'",
+            ),
+            (
+                # Specify a sort using a bad metadata namespace
+                "xyzzy.uploaded",
+                "Metadata key 'xyzzy.uploaded' is not supported",
+            ),
+        ],
+    )
+    def test_dataset_sort_errors(self, server_config, query_as, sort, message):
+        """Test `datasets/list?sort` error cases
+
+        Args:
+            server_config: The PbenchServerConfig object
+            query_as: A fixture to provide a helper that executes the API call
+            sort: A JSON representation of the sort query parameter value
+            message: The expected error message
+        """
+
+        # Assign "sequence numbers" in the inverse order of name
+        test = User.query(username="test")
+        all = Database.db_session.query(Dataset).order_by(desc(Dataset.name)).all()
+        for i, d in enumerate(all):
+            odd = i & 1
+            Metadata.setvalue(d, "global.test.sequence", i)
+            Metadata.setvalue(d, "user.test.odd", odd, user=test)
```

Comment:
This, of course, isn't necessary for the error case. I meant to remove it and forgot. Hardly a crisis; but I'll dump these lines if I'm given reason to make another commit.

Reply:
> if I'm given reason to make another commit.

Did I come through for you, Dave?  😆